src/module/auth/application/service/auth.service.ts

```diff
@@ -148,21 +147,23 @@ export class AuthService {
     return accessToken;
   }
 
-  // async refreshToken(refreshToken: string): Promise<{ accessToken: string }> {
-  //   await this.authRepository.removeRefreshToken(refreshToken);
-
-  //   let userToRefresh: any;
-
-  //   const verify = await this.jwtService.verify(
-  //     refreshToken,
-  //     this.REFRESH_TOKEN_SECRET,
-  //   );
-
-  //   if (verify) {
-  //     const user = await this.userService.getUserByEmail(userToRefresh.email);
-  //     const accessToken = this.getAccessToken(user);
-
-  //     return { accessToken };
-  //   }
-  // }
+  async refreshToken(refreshToken: string): Promise<string> {
+    try {
+      await this.authRepository.removeRefreshToken(refreshToken);
+
+      const verify = this.jwtService.verify(refreshToken, {
+        secret: this.REFRESH_TOKEN_SECRET,
+      });
+
+      if (verify) {
+        const user = await this.userService.getUserByEmail(verify.email);
+        const accessToken = await this.getRefreshToken(user);
+        const newAuth = new Auth(accessToken, user);
+        await this.authRepository.saveRefreshToken(newAuth);
+        return accessToken;
+      }
+    } catch (error) {
```

Comment:
I think this try-catch is not necessary, this code came from the express app, In the express app was needed

Reply:
I have been reviewing how to remove this try-catch, but JWT throws a token verification error when receiving invalid tokens. Now, I am handling that error from the catch block. I can't seem to find the class to which this error belongs to throw a more appropriate error.